### PR TITLE
feat(ant-stack): root dev server

### PR DESCRIPTION
**Pull request by ap0nia**
TODO:
- [x] Every single endpoint is a subproject that should be individually built by ESBuild to a `dist/index.js`
- [x] Keep track of all of these projects in an `Object` or `Map`
- [x] For each `dist/index.js`, create a new middleware router just for that route to enable hot-swapping per endpoint
- [x] Chokidar needs to watch all of the discovered sub-project files
- [x] Whenever a file changes, figure out which sub-project (i.e. endpoint) it belongs to, and recompile + re-assign the middleware router for that endpoint

### CODE REVIEW

**Comment by ap0nia** [comment]
Oh oops, I have some of the old code from your create branch. How do I update it?

**Comment by ecxyzzy** [comment]
You wanna have #48 merged into this branch and then merge the whole thing into `main`?

**Comment by ap0nia** [comment]
Ok sure!

File: packages/ant-stack/src/cli/commands/dev.ts [hunk 1]
```diff
@@ -84,3 +106,146 @@ export async function startDevServer() {
...
+    });
+  };
+
+  /**
+   * Prepare the development server by loading all the endpoints and refreshing the routes.
+   */
+  await Promise.all(endpoints.map(loadEndpoint)).then(refreshRouter);
+
+  app.listen(config.port, () => {
+    consola.log(`🚀 Express server listening at http://localhost:${config.port}`);
```
Comment (ecxyzzy) [inline packages/ant-stack/src/cli/commands/dev.ts:232]:
```suggestion
    consola.info(`🚀 Express server listening at http://localhost:${config.port}`);
```

File: packages/ant-stack/src/cli/commands/dev.ts [hunk 2]
```diff
@@ -84,3 +106,146 @@ export async function startDevServer() {
...
+        endpointMiddleware[endpoint](req, res, next)
+      );
+    });
+  };
+
+  /**
+   * Load a specific endpoint's middleware.
+   */
+  const loadEndpoint = async (endpoint: string) => {
+    console.log("setting up router for ", endpoint);
```
Comment (ecxyzzy) [inline packages/ant-stack/src/cli/commands/dev.ts:204]:
```suggestion
    consola.info(`🚀 Setting up router for ${endpoint}`);
```

**Review by ecxyzzy — CHANGES_REQUESTED** [review]
Looks good. One minor issue, aside from logging consistency suggestions:

`findAllProjects` seems to double count the `/v1/rest/klein` endpoint and only that endpoint; not sure why. Can you recreate this, and if so is the solution literally just tossing it into a `Set` to dedupe?

**Comment by ap0nia** [comment]
Oh, yeah I was wondering why it would load some endpoint twice lol. Yeah let me use a set

**Review by ecxyzzy — APPROVED** [review]
LGTM now, awesome job!!

**Pull request merged**